sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -0,0 +1,218 @@
+package io.sentry;
+
+import io.sentry.protocol.Contexts;
+import io.sentry.protocol.SentryId;
+import io.sentry.util.Objects;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class SentryTransaction extends SentryBaseEvent implements ISpan {
+  /** The transaction name. */
+  private @Nullable String transaction;
+
+  /** The moment in time when span was started. */
+  private final @NotNull Date startTimestamp;
+
+  /** The moment in time when span has ended. */
+  private @Nullable Date timestamp;
```

Comment:
Java Date only have millis precision, is that the intended behavior? do we need to optimize more than that?

Reply:
According to https://develop.sentry.dev/sdk/event-payloads/span/ millis precision is expected.

Reply:
thanks